text/0033-view-hierarchy.md

```diff
@@ -0,0 +1,56 @@
+* Start Date: 2022-11-02
+* RFC Type: feature
+* RFC PR: [#33](https://github.com/getsentry/rfcs/pull/33)
+* RFC Status: -
+* RFC Driver: [Dhiogo Ramos Brustolin](https://github.com/brustolin)
+
+# Summary
+
+Allow SDKs to send a textual representation of the application's UI view hierarchy. 
+
+# Motivation
+
+Being able to see what the user was looking at is a good way to found out problems in the application. 
+This RFC proposes a way to send information from the UI state of an application, known as "view hierarchy".
+
+# Proposal 
+
+The proposal is to create a new [envelope item](https://develop.sentry.dev/sdk/envelopes/) in the form of a json representation of the UI elements. 
+This is how the envelope item header should look like:
+
+```json
+{"type":"view_hierarchy","length":...,"content_type":"application/json"}\n
+```
+
+And the envelope item payload will be a JSON array that contains a view hierarchy for each available screen/window. 
+This is how it should look like: 
+
+```json
+[
+    {
+        "type": "element_class_name", 
+        "identifier": "element_identifier",
+        "children": [...], //An array of ui elements.
+        "width": 100, 
+        "height": 100,
+        "depth": 100, //if 3D
+        "x": 0,
+        "y": 1,
+        "z": 2, //if 3D.
+        "visible": true|false,
+        "alpha": 1,
+        "{extra_properties}": "{property value}" //adicional information by platform
```

Comment:
What if we make a databag here, `Map<String, Any>`?

Reply:
It's possible. I'm not a fan. 
Property names can't duplicate, I don't see a reason we need an extra layer of properties.

Reply:
No strong opinions, at least on Relay, we often have the pattern of this `data` databad that holds extra values.

Reply:
I'd prefer being able to separate out dynamically added properties (i.e. between platforms) from common ones. I'm still trying to understand how this data will get surfaced in the product after ingestion, but knowing that certain keys are always there vs not will be helpful when working with the data type in the front end

Reply:
I think we should always all for forwards compatibility for event schemas, so we allow for arbitrary keys, but we don't do anything with them.